internal/api/grpc_service.go

```diff
@@ -443,6 +442,62 @@ func (s *GrpcService) GetDeviceDefinitionAll(ctx context.Context, in *emptypb.Em
 	return result, nil
 }
 
+func (s *GrpcService) GetDeviceCompatibility(ctx context.Context, in *p_grpc.GetDeviceCompatibilityListRequest) (*p_grpc.GetDeviceCompatibilityListResponse, error) {
+	qryResult, _ := s.Mediator.Send(ctx, &queries.GetDeviceCompatibilityQuery{
+		MakeID:        in.MakeId,
+		IntegrationID: in.IntegrationId,
+		Region:        in.Region,
+	})
+
+	deviceCompatibilities := qryResult.(queries.GetDeviceCompatibilityQueryResult)
+
+	result := &p_grpc.GetDeviceCompatibilityListResponse{}
+
+	integFeats := deviceCompatibilities.IntegrationFeatures
+	dcMap := make(map[string][]*p_grpc.DeviceModelYears)
+	for _, v := range deviceCompatibilities.DeviceDefinitions {
+		// struct has noa values yet
+		if _, ok := dcMap[v.Model]; !ok {
+			dcMap[v.Model] = []*p_grpc.DeviceModelYears{}
+		}
+		if len(v.R.DeviceIntegrations) == 0 {
+			continue
+		}
+
+		di := v.R.DeviceIntegrations[0]
+
+		if di.Features.IsZero() {
+			continue
+		}
```

Comment:
We shouldn't hit any of these because of the query, yeah? Maybe worth a warning?